blazor/dropdown-list/data-binding.md

```diff
@@ -7,68 +7,378 @@ control: DropDown List
+The DropDownList loads the data from local data sources through the DataSource property. It supports the data type of Array, Observable Collection, ExpandoObject, DynamicObject.
 
-@code {
+{% highlight cshtml %}
 
-    public string DropVal;
+{% include_relative code-snippet/data-binding/local-data-binding.razor %}
 
-    public class Country
-    {
-        public string Name { get; set; }
+{% endhighlight %}
 
-        public string Code { get; set; }
-    }
+![Blazor DropdownList with local data binding](./images/data-binding/blazor_dropdown_local-binding.png)
+
+### Primitive type
+
+You can bind the data to the DropDownList as a list of string, int, double and bool type items.
+
+The following code demonstrates array of string values to the DropDownList component.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/primitive-type-string.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Primitive string type](./images/data-binding/blazor_dropdown_primitive-type-string.png)
+
+The following code demonstrates array of integer values to the DropDownList component.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/primitive-type-int.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Primitive int type](./images/data-binding/blazor_dropdown_primitive-type-int.png)
+
+### Complex data type
+
+The DropDownList can generate its list items through an array of complex data. For this, the appropriate columns should be mapped to the [Fields](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.DropDowns.FieldSettingsModel.html) property.
+
+In the following example, `Code.ID` column and `Country.CountryID` column from complex data have been mapped to the [`DropDownListFieldSettings.Value`](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.DropDowns.FieldSettingsModel.html#Syncfusion_Blazor_DropDowns_FieldSettingsModel_Value) and  [`DropDownListFieldSettings.Text`](https://help.syncfusion.com/cr/blazor/Syncfusion.Blazor.DropDowns.FieldSettingsModel.html#Syncfusion_Blazor_DropDowns_FieldSettingsModel_Text), respectively.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/complex-data-type.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Complex data type](./images/data-binding/blazor_dropdown_complex-data-type.png)
+
+### Expando object binding
+
+You can bind ExpandoObject data to the DropDownList component. The following example `ExpandoObject` is bound to the collection of vehicles data.
+
+{% highlight cshtml %}
+
+{% include_relative code-snippet/data-binding/expando-object-binding.razor %}
+
+{% endhighlight %}
+
+![Blazor DropdownList with Expando object binding](./images/data-binding/blazor_dropdown_expando-object-binding.png)
+
+### Observable collection binding
+
+You can bind ObservableCollection data to the DropDownList component. In the following example, `Observable Data` is bound to a collection of colors data.
```

Comment:
Link the common topic - https://blazor.syncfusion.com/documentation/common/data-binding/data-updates#observable-collection